### ISSUE TITLE
How to import categorylinks and page_props seperately in english wikipedia?

### ISSUE THREAD

**Issue by AntiMortal** [gnosygnu/xowa#757]
Is there any way to somehow seperately import that two files after I imported wikipedia and pagelinks files? I don't want to restart the whole process which would take about 9 hours. I am scared that wikipedia will be removed and reimported. I made all xowa files read-only. Can I somehow skip reimporting main wikipedia file (enwiki-20200620-pages-articles)?

**Comment by gnosygnu** [comment]
Hi! Thanks for the interest in XOWA!

You can rebuild the category system for a wiki. I haven't tried it for a while, but I kicked off the process and it looks like it's running fine. Basically, do the following:
* Download https://dumps.wikimedia.org/enwiki/latest/enwiki-latest-page_props.sql.gz and https://dumps.wikimedia.org/enwiki/latest/enwiki-latest-categorylinks.sql.gz
* Unzip them to your en.wikipedia.org folder. For example, you may have files like the following when done:
  * D:\xowa\wiki\en.wikipedia.org\enwiki-latest-categorylinks.sql
  * D:\xowa\wiki\en.wikipedia.org\enwiki-latest-page_props.sql
* Open up XOWA and go to this page: home/wiki/Dashboard/Import/Offline
* Copy and paste the following into the main box:
```
// rebuild category
app.setup.cmds.dump_add_many( 'en.wikipedia.org', 'latest', '', 'wiki.category2.build');
```
* Press `Run Script` and wait for a few hours

Note that you will need to mark all the `en.wikipedia.org-core.xowa` file read-write. Unfortunately, rebuilding is going to change this file. Just make a copy of the file if you're worried something will go wrong. 

If you want a little more control, you can look at the command-line version here: http://xowa.org/home/wiki/Dev/Command-line/Dumps . Word of warning though: it's pretty complicated

Hope this helps!

**Comment by AntiMortal** [comment]
I am still in the process of downloading categorylinks, and will probably be able to try in a day or two.
I think I already tried that, as it said so on Wiki Maintenance, and it did not work maybe because core file was read only or word latest had to be switched to proper word in file name or vice versa, for me it is 20200620.
Can I leave text file ready only and are they affected too? Do I have to make backup of them too?

Also one small question not related to this: is cache keeping keeping files permanently or do they expire eventually?
I mean files made by allowing xowa to connect online while reading offline wikipedia. Those files in 'file' folder and files en.wikipedia.org-file-core.xowa, en.wikipedia.org-file-user.xowa. Of course I don't keep them read only.

**Comment by AntiMortal** [comment]
OK, thanks, it worked. Now I have categories, although it took almost whole day to import them. And I noticed the only file that got modified of old files is core. Text and search were left untouched. I won't close this issue in case you decide to add gui ability to import search and categories seperately without downloading again, similar to wiki maintenance option, just with download commands skipped.

**Comment by gnosygnu** [comment]
> OK, thanks, it worked.

Cool. Thanks for the confirmation.

> And I noticed the only file that got modified of old files is core. Text and search were left untouched.

Yup, that's correct.

> I won't close this issue in case you decide to add gui ability to import search and categories seperately without downloading again, similar to wiki maintenance option, just with download commands skipped.

By Search, do you mean full-text search? If so, there's a way to rebuild them but it requires a different page. See: http://xowa.org/home/wiki/App/Full-text_search/Lucene/Search_indexes/Building

**Comment by AntiMortal** [comment]
No, just title search. I mean you should add ability to import categories and search seperately offline, without downloading.
Maybe in in import offline page, or maybe in wiki maintenance page.
Actually looks like search button in wiki maintenance page already exists, but category button is for download+import.
Then maybe you should add additional button to just import category, without downloading.
Although for me personally commands are self-explanatory, so I now know that removing two download commands is for importing categories only, those commands made by clicking category button in wiki maintenance.
Search button is for just importing pagevews?
